### PR TITLE
WIP: FI-610 create validators from the FHIR Models Expansions JSON

### PR DESCRIPTION
**Pull request by okeefm**
This pull request adds a pair of methods to get additional valuesets and codesystems into the terminology service, by loading expansions from FHIR::Models. It also parses the loaded valuesets to create Bloom Filters for codesystems.

Pull requests into Inferno require the following items to be completed. Submitter and reviewer 
should check the relevant check boxes when the associated item is done. For items that are not 
applicable, note it's not applicable ("N/A") and check the box. For example, external Pull 
Requests do not require ticket links.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-610
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/app/utils/terminology.rb [hunk 1]
```diff
@@ -145,6 +152,78 @@ def self.load_validators(directory = 'resources/terminology/validators/bloom')
...
+      Inferno.logger.debug 'Loading FHIR Models Expansions'
+      FHIR::Definitions.expansions.each do |expansion|
+        url = expansion['url']
+        next if @known_valuesets[url]
+        next if SKIP_SYS.include? url
+
+        valueset = Inferno::Terminology::Valueset.new(@db)
+        valueset.valueset_model = FHIR::ValueSet.new(expansion)
+        valueset.vsa = self
+        if valueset&.valueset_model&.expansion&.contains
```
[the remark this answers is not in the document]
Comment (okeefm) [reply lib/app/utils/terminology.rb]: Sounds like a good improvement. I'll do that.

File: lib/app/utils/terminology.rb [hunk 2]
```diff
@@ -145,6 +152,78 @@ def self.load_validators(directory = 'resources/terminology/validators/bloom')
...
+        valueset.valueset.each do |coding|
+          codesystem_bloomfilters[coding[:system]] ||= Bloomer::Scalable.new
+          codesystem_bloomfilters[coding[:system]].add coding
+        end
+      end
+
+      codesystem_bloomfilters.each do |codesystem, bfilter|
+        Inferno.logger.debug "Processing codesystem #{codesystem}"
+        next if @loaded_validators[codesystem]
+        next unless codesystem
```
[the remark this answers is not in the document]
Comment (okeefm) [reply lib/app/utils/terminology.rb]: Good catch.

File: lib/app/utils/terminology.rb [hunk 3]
```diff
@@ -145,6 +152,75 @@ def self.load_validators(directory = 'resources/terminology/validators/bloom')
...
+      FHIR::Definitions.expansions.each do |expansion|
+        url = expansion['url']
+        next if @known_valuesets[url]
+        next if SKIP_SYS.include? url
+
+        valueset = Inferno::Terminology::Valueset.new(@db)
+        valueset.valueset_model = FHIR::ValueSet.new(expansion)
+        valueset.vsa = self
+
+        @known_valuesets[valueset.url] = valueset
```
Comment (radamson) [inline lib/app/utils/terminology.rb]: nothing below this line is needed.  The `create_validators` call in the rake task will take care of it.

File: lib/app/utils/terminology.rb [hunk 4]
```diff
@@ -145,6 +152,75 @@ def self.load_validators(directory = 'resources/terminology/validators/bloom')
...
+          bfilter.include? probe
+        end
+        # Register the validators with FHIR Models for validation
+        FHIR::DSTU2::StructureDefinition.validates_vs(url, &validate_fn)
+        FHIR::StructureDefinition.validates_vs(url, &validate_fn)
+        @loaded_validators[url] = expansion['expansion']['total']
+      end
+    end
+
+    def self.parse_codesystems_from_valuesets
```
Comment (radamson) [inline lib/app/utils/terminology.rb]: This probably won't provide a full expansion of the codesystem, so lets remove this for now.  We'd really have to inspect the compose to make sure the full codesystem is used in order to be sure.

Comment (okeefm) [reply lib/app/utils/terminology.rb]: Removed

**Pull request closed without merging**